app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -247,16 +305,159 @@ fun ServiceRequestItem(request: ServiceRequest) {
+                      packageProposal = packages[index],
+                      isSelected = isSelected,
+                      modifier =
+                          Modifier.width(260.dp) // Slightly wider for better touch targets
+                              .height(size)
+                              .clickable { selectedIndex = if (isSelected) -1 else index }
+                              .testTag("PackageCard"),
+                      selectedPackage = selectedPackage)
+                }
+              }
+              Spacer(modifier = Modifier.height(16.dp)) // Add space between LazyRow and Button
+              Button(
+                  onClick = {
+                    selectedPackage.value?.let {
+                      requestViewModel.saveServiceRequest(
+                          request.copy(
+                              providerId = providerId,
+                              packageId = it.uid,
+                              status = ServiceRequestStatus.ACCEPTED))
+                      showDialog.value = false
+                    }
+                  },
+                  enabled = selectedPackage.value != null) {
+                    Text(text = "Propose Package")
+                  }
+            } else {
+              Card(
+                  modifier = Modifier.fillMaxWidth().padding(16.dp),
+                  elevation = CardDefaults.cardElevation(8.dp),
+                  colors = CardDefaults.cardColors(containerColor = colorScheme.primaryContainer)) {
+                    Column(
+                        modifier = Modifier.padding(16.dp).fillMaxWidth(),
+                        horizontalAlignment = Alignment.CenterHorizontally) {
+                          Text(
+                              text =
+                                  "No packages available. Please enter a price for this service:",
+                              fontSize = 18.sp,
+                              fontWeight = FontWeight.Bold,
+                              color = colorScheme.onPrimaryContainer)
+                          Spacer(modifier = Modifier.height(16.dp))
+                          var priceInput by remember { mutableStateOf("") }
+                          val isPriceValid = priceInput.matches(Regex("^[0-9]+(\\.[0-9]{0,2})?$"))
+                          TextField(
+                              value = priceInput,
+                              onValueChange = { input ->
+                                if (input.matches(Regex("^[0-9]*(\\.[0-9]{0,2})?$"))) {
+                                  priceInput = input
+                                }
+                              },
+                              label = { Text("Price") },
+                              modifier = Modifier.fillMaxWidth().testTag("PriceInput"),
+                              keyboardOptions =
+                                  KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number),
+                              singleLine = true,
```

Comment:
We can think to add an invalidation input to tell the user that he can't type it in this way

Reply:
I can also add this kind of prices ! Should we keep it ??
<img width="195" alt="image" src="https://github.com/user-attachments/assets/7dc0572c-b41a-404a-ac9c-f61e7f0398f6">

Reply:
I can implement it such that leading 0 are discarded, would that be satisfactory? I can also limit the numbers to let's say 10 digits?